pkg/v1/tkg/tkgpackageclient/image_utils.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tkgpackageclient
+
+import (
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+
+	kappipkg "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgpackagedatamodel"
+)
+
+// parseRegistryImageURL parses the registry image URL to get repository and tag, tag is empty if not specified
+func parseRegistryImageURL(imgURL string) (repository, tag string, err error) {
+	ref, err := name.NewTag(imgURL)
+	if err != nil {
+		return "", "", err
+	}
+
+	repository = ref.Context().String()
+	tag = ref.TagStr()
+	// dockerParser sets the tag to "latest" if not specified, however we want it to be empty
+	if tag == tkgpackagedatamodel.DefaultRepositoryImageTag && !strings.HasSuffix(imgURL, ":"+tkgpackagedatamodel.DefaultRepositoryImageTag) {
+		tag = ""
+	}
+	return repository, tag, nil
+}
+
+// GetCurrentRepositoryAndTagInUse fetches the current tag used by package repository, taking tagselection into account
+func GetCurrentRepositoryAndTagInUse(pkgr *kappipkg.PackageRepository) (repository, tag string, err error) {
+	if pkgr.Spec.Fetch == nil || pkgr.Spec.Fetch.ImgpkgBundle == nil {
+		return "", "", errors.New("failed to find OCI registry URL")
+	}
+
+	repository, tag, err = parseRegistryImageURL(pkgr.Spec.Fetch.ImgpkgBundle.Image)
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to parse OCI registry URL")
+	}
+
+	if tag == "" {
+		tag = tkgpackagedatamodel.LatestReleaseTag
```

Comment:
We are treating empty tag as `LATEST RELEASE` here and `Constraints:>0.0.0` for creating new repository below. But they are not exactly the same, for instance 0.1.0 can satisfy >0.0.0 but latest is probably another tag. cc @cppforlife

Reply:
Actually I'm not sure what would be the right word to put here.
My logic is that `Constraints:>0.0.0` will select the latest non-prerelease version, so I call the tag `LATEST RELEASE  `.

Reply:
Offline sync with @blc1996 and got confirmation that kapp-controller with `Constraints:>0.0.0` will pull a newer version if the current version selected is no longer latest so `Constraints:>0.0.0` ~ `latest`.